File: Problems 2100-2199/2109. Adding Spaces to a String/solution.cpp
class Solution {
public:
    string addSpaces(string s, vector<int>& spaces) {
        string res(s.length() + spaces.size(), ' ');
        int j = 0;
        for (int i = 0; i < s.length(); i++) {
            if (j < spaces.size() && spaces[j] == i) {
                j++;
            }
            res[i+j] = s[i];
        }
        return res;
    }
};

File: Problems 2100-2199/2109. Adding Spaces to a String/solution.py
class Solution:
    def addSpaces(self, s: str, spaces: List[int]) -> str:
        m, n = len(s), len(spaces)
        res = [' '] * (m + n)
        j = 0
        for i in range(m):
            if j < n and spaces[j] == i:
                j += 1
            res[i+j] = s[i]
        
        return "".join(res)